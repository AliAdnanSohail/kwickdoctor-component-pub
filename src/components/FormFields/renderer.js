import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
