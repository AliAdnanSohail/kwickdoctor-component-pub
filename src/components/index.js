export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
