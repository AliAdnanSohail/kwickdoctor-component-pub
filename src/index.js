export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
