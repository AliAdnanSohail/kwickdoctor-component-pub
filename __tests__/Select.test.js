import React from 'react';
import renderer from 'react-test-renderer';

import { Select } from '../src';

const defaultProps = {
  name: 'gender',
  selected: 'any',
  options: [{ value: 'one' }, { value: 'two' }, { value: 'three' }, { value: 'four' }],
  label: 'Label',
};

it('renders correctly', () => {
  const element = renderer.create(<Select {...defaultProps} />).toJSON();

  expect(element).toMatchSnapshot();
});
