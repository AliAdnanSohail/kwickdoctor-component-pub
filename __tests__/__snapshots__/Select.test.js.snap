// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="jsx-1685083184 jsx-4038665278 jsx-2881818901 form-field"
>
  <label
    className="jsx-1685083184 jsx-4038665278 jsx-2881818901"
    htmlFor={undefined}
  >
    Label
  </label>
  <div
    className="select-container"
  >
    <select
      className=""
      id={undefined}
      name="gender"
      placeholder={undefined}
      type={undefined}
      value={undefined}
    >
      <option
        value="male"
      >
        male
      </option>
      <option
        value="any"
      >
        any
      </option>
      <option
        value="female"
      >
        female
      </option>
    </select>
    <svg
      aria-label="down"
      className="grommetux-control-icon grommetux-control-icon-down grommetux-control-icon--responsive"
      height="24px"
      role="img"
      version="1.1"
      viewBox="0 0 24 24"
      width="24px"
    >
      <polyline
        fill="none"
        points="7.086 3.174 17.086 13.174 7.086 23.174"
        stroke="#000"
        strokeWidth="2"
        transform="scale(1 -1) rotate(-89 -1.32 0)"
      />
    </svg>
  </div>
  <style>
    input,textarea,select{width:100%;min-width:100%;max-width:100%;min-height:44px;padding:12px 16px;background-color:#ffffff;color:#2A2A30;box-sizing:border-box;-webkit-appearance:none;font-style:normal;font-weight:normal;font-size:0.875rem;border-width:2px;border-style:solid;border-color:#E7E7F0;border-radius:4px;}
    input::-webkit-input-placeholder,textarea::-webkit-input-placeholder,select::-webkit-input-placeholder{color:#828393;}
    input::-moz-placeholder,textarea::-moz-placeholder,select::-moz-placeholder{color:#828393;}
    input:-ms-input-placeholder,textarea:-ms-input-placeholder,select:-ms-input-placeholder{color:#828393;}
    input:-moz-placeholder,textarea:-moz-placeholder,select:-moz-placeholder{color:#828393;}
    input:focus,textarea:focus,select:focus{outline:none;padding:12px 16px;border-color:#0c97f9;}
    input.invalid,input.invalid:focus,textarea.invalid,textarea.invalid:focus,select.invalid,select.invalid:focus{border-color:#FF5572;}
  </style>
  <style>
    .select-container{position:relative;}
    .select-container select{padding-right:30px;}
    .select-container svg{position:absolute;right:15px;top:calc(50% - 6px);height:12px;width:12px;fill:#E7E7F0;stroke:#E7E7F0;}
    .select-container svg polyline{stroke-width:5;}
  </style>
</div>
`;
